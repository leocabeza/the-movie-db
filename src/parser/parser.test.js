import parseData, { setDataAccordingToValueType } from './parser';
import popularMoviesMockResponse from './mock.json';

describe('parser', () => {
  it('must parse correctly a mocked response', () => {
    const parsedResponse = parseData(popularMoviesMockResponse);

    expect(parsedResponse).toMatchSnapshot();
  });

  it('must set the same data type that it was passed in key', () => {
    const stringData = { name: 'Community' };
    const arrayData = { ids: [1, 2] };
    const objectData = { metadata: { producer: 'Abed Nadir' } };

    expect(typeof setDataAccordingToValueType(stringData, 'name')).toEqual(
      typeof stringData['name']
    );
    expect(
      Array.isArray(setDataAccordingToValueType(arrayData, 'ids'))
    ).toBeTruthy();
    expect(setDataAccordingToValueType(objectData, 'metadata')).toBeInstanceOf(
      Object
    );
  });

  it('must parse data correcty', () => {
    const emptyData = [];
    const dataWithKey = {
      writer_producer: 'Dan Harmon',
      seasons: [{ name: 'Season 1' }],
    };

    expect(parseData(emptyData)).toEqual([]);
    expect(parseData(dataWithKey)).toEqual(dataWithKey);
  });
});
