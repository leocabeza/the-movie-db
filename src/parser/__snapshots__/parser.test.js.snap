// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser must parse correctly a mocked response 1`] = `
Object {
  "page": 1,
  "results": Array [
    Object {
      "adult": false,
      "backdrop_path": "/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg",
      "genre_ids": Array [
        878,
      ],
      "id": 335983,
      "original_language": "en",
      "original_title": "Venom",
      "overview": "When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego \\"Venom\\" to save his life.",
      "popularity": 240.47,
      "poster_path": "/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg",
      "release_date": "2018-10-03",
      "title": "Venom",
      "video": false,
      "vote_average": 6.6,
      "vote_count": 2113,
    },
  ],
  "total_pages": 993,
  "total_results": 19847,
}
`;
